### DISCUSSION TITLE
Is everything truly open-source or is it more like an "open paywall"?

### DISCUSSION THREAD

**Discussion by nyxb** [nhost/nhost#2891] [Q&A]
Hello Nhost team,

I've been exploring your platform and noticed that it's advertised as open-source. I wanted to clarify if all the code from the cloud version is available. Specifically, would it be possible, theoretically, to use your codebase to set up an identical self-hosted service, including server setups similar to those running on AWS?

I'm curious if this is fully open-source or if there are limitations like those seen with Supabase, where while it's labeled "open-source," certain features are behind what some refer to as an "open paywall."

Thank you for any clarification!

Best regards,
nyxb

**Comment by dbarrosop** [comment]
Hello,
I have never heard the term "open paywall" and I am also unsure to which supabase features you are referring to so I am not sure how to answer. In any case, the auth/storage services we use in the cloud for our customers are the same that you can find in our github organization and those are Apache2 or MIT license. Postgres is vanilla postgres and hasura is the community edition. Obviously this doesn't extend to our cloud solution; orchestrators, observability, management, etc. None of that is opensource but those are purely control plane services, they don't add any user-facing functionality.

The only exception to the above is our [AI service](https://docs.nhost.io/product/ai) which is an additional service we added recently and that we haven't opensourced yet and requires a license if you are self-hosting.

Regards.

**Reply by nyxb** [reply]
and why aren't these controls also opensource? The ai service needs a license from you or from the ai api service? Openpaywall is a new term, it describes something that is not entirely opensource. 

Best regards

**Reply by dbarrosop** [reply]
searching online for "open paywall" didn't return any results so if you could provide some article or something explaining the term that would be ideal. In any case, regardless of terms people like to throw around, the only thing that matters is the license and as mentioned previously all of our services (auth, storage, graphql and database) are fully covered by OSI approved licenses.

Our control plane isn't opensource because that is how we manage our cloud.

For the AI service you need a license from us but the AI service isn't part of the Nhost stack. It is an extra service you have to add to it if you want to.
